worlds/marioland2/options.py

```diff
@@ -0,0 +1,165 @@
+from Options import Toggle, Choice, NamedRange, Range, PerGameCommonOptions
+from dataclasses import dataclass
+
+
+class ShuffleGoldenCoins(Choice):
+    """Vanilla: Golden Coins are not in the item pool, you receive them when defeating bosses as in vanilla.
+    Shuffle: Shuffle the Golden Coins into the item pool and open location checks on bosses.
+    Mario Coin Fragment Hunt: You start with all Golden Coins except the Mario Coin, which has been fragmented into
+    many pieces.
+    You will see a Golden Coin being received when defeating bosses regardless of whether you are actually getting a coin."""
+    display_name = "Shuffle Golden Coins"
+    default = 0
+    option_vanilla = 0
+    option_shuffle = 1
+    option_mario_coin_fragment_hunt = 2
+
+
+class GoldenCoinsRequired(Range):
+    """Number of Golden Coins required to enter Mario's Castle. Ignored on Mario Coin Fragment Hunt."""
+    display_name = "Golden Coins Required"
+    range_start = 0
+    range_end = 6
+    default = 6
+
+
+class MarioCoinFragmentPercentage(Range):
+    """Percentage of filler items to be replaced with Mario Coin Fragments. Note that the Coinsanity and Coinsanity
+    Checks settings will greatly impact the number of replaceable filler items. There may be as few as 6 available
+    slots for Mario Coin Fragments if Coinsanity is off."""
+    display_name = "Mario Coin Fragment Percentage"
+    range_start = 1
+    range_end = 100
+    default = 20
+
+
+class MarioCoinFragmentsRequiredPercentage(Range):
+    """Percentage of the Mario Coins in the item pool that are required to put the Mario Coin together."""
+    display_name = "Mario Coin Fragments Required Percentage"
+    range_start = 1
+    range_end = 100
+    default = 75
+
+
+class ShuffleMidwayBells(Toggle):
+    """Shuffle the Midway Bells into the item pool. Ringing a bell will trigger a location check.
+    Obtaining a Midway Bell will be permanent, and some levels will require backtracking from the midway point to reach
+    secret exits."""
+    display_name = "Shuffle Midway Bells"
+
+
+class MariosCastleMidwayBell(Toggle):
+    """Adds a Midway Bell to the final stage, just before the Wario fight."""
+    display_name = "Mario's Castle Midway Bell"
+
+
+class Coinsanity(Toggle):
+    """Shuffles the singular coins found freestanding and in question mark blocks into the item pool, and adds location
+    checks made by obtaining a sufficient number of coins in particular levels within a single playthrough."""
+    default_name = "Coinsanity"
+
+
+class CoinsanityChecks(Range):
+    """Number of Coinsanity checks.
+    A higher number means more checks, and smaller coin amounts per coin item in the item pool.
+    If Accessibility is set to Locations, auto-scroll levels may have a lower maximum count, which may lead to this
```

Comment:
This line will become outdated with #1481

Reply:
Changed